src/App.tsx

```diff
@@ -8,37 +8,159 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { User } from './types/User';
+import { Post } from './types/Post';
+import { Comment, CommentData } from './types/Comment';
+import { client } from './utils/fetchClient';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [currentPost, setCurrentPost] = useState<Post | null>(null);
+  const [loadingPosts, setLoadingPosts] = useState(false);
```

Comment:
Fix it everywhere with [tips on naming boolean variables](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)
```suggestion
  const [isLoadingPosts, setIsLoadingPosts] = useState(false);
```